.vitepress/theme/components/BannerComponent.vue

```diff
@@ -0,0 +1,113 @@
+<script setup lang="ts">
+import { useElementSize } from '@vueuse/core';
+import { onBeforeMount, onMounted, ref, watchEffect } from 'vue';
+
+const el = ref<HTMLElement>();
+const { height } = useElementSize(el);
+
+watchEffect(() => {
+  if (height.value) {
+    document.documentElement.style.setProperty(
+      '--vp-layout-top-height',
+      `${height.value + 16}px`
+    );
+  }
+});
+
+const dismiss = () => {
+  localStorage.setItem(
+    'dismissBanner',
+    (Date.now() + 8.64e7 * 1).toString() // current time + 1 day
+  );
+  document.documentElement.classList.add('banner-dismissed');
+};
+
+onBeforeMount(() => {
+  if (localStorage.getItem('dismissBanner') !== null) {
+    var date = parseInt(localStorage.getItem('dismissBanner')!);
```

Comment:
Nit, store the local stotrage item in a variable, there is a possible (unlikely) race condition here.